File: 621.py
from collections import Counter

class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        counts = tuple(x[1] for x in Counter(tasks).most_common())
        chunk = counts[0] - 1
        idleCount = chunk * n
        for idx in range(1, len(counts)):
            idleCount -= min(chunk, counts[idx])
            if idleCount <= 0:
                break
        return len(tasks) + max(0, idleCount)

File: 621.cpp
struct Solution {
    int leastInterval(vector<char>& tasks, int n) {
        int freq[26] = {0};
        for (auto c : tasks){
            ++freq[c - 'A'];
        }
        sort(freq, freq + 26, std::greater{});
        int chunk = freq[0] - 1;
        int idleCount = chunk * n;
        for (int i = 1; i < 26; ++i){
            if (freq[i] == 0){
                break;
            }
            idleCount -= min(chunk, freq[i]);
            if (idleCount <= 0){
                break;
            }
        }
        return tasks.size() + max(0, idleCount);
    }
};
